purchase_security/models/res_partner.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    # add indexes for better performance on record rules
+    user_id = fields.Many2one(index=True)
+
+    def _remove_key_followers(self, partner):
+        for record in self.mapped("commercial_partner_id"):
+            record.message_unsubscribe(partner_ids=partner.ids)
+
+    def _add_followers_from_salesmans(self):
```

Comment:
I don't think you should propagate followers, unless you add a special new field in `res.partner` being "Purchase representative", distinct from the salesman. If not, you are mixing boundaries using the same field for both purchase/sales.